### ISSUE TITLE
Storybook server does not start if error in initial build, even after fixing

### ISSUE THREAD

**Issue by danawoodman** [storybookjs/storybook#9889]
**Describe the bug**
The Storybook server running on localhost:6006 does not start if  you startup storybook with initial errors in your JS. Even after fixing, the server never manages to start so you have to kill and restart Storybook. 

**To Reproduce**
1. Start storybook with some JS error in your Story source/imported JS
2. Storybook fails to start (expected)
3. Fix error
4. Storybook build starts but server never starts up (eg localhost:6006 is not running)
5. Have to kill and restart storybook for it to work

**Expected behavior**
Server to start after initial error is fixed.

**Screenshots**
n/a

**Code snippets**
n/a

**System:**
```
Environment Info:

  System:
    OS: macOS 10.15.3
    CPU: (12) x64 Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
  Binaries:
    Node: 13.8.0 - ~/.nvm/versions/node/v13.8.0/bin/node
    npm: 6.13.6 - ~/.nvm/versions/node/v13.8.0/bin/npm
  Browsers:
    Chrome: 80.0.3987.106
    Firefox: 64.0
    Safari: 13.0.5
  npmPackages:
    @storybook/addon-actions: 5.3.12 => 5.3.12
    @storybook/addon-info: 5.3.12 => 5.3.12
    @storybook/addon-knobs: 5.3.12 => 5.3.12
    @storybook/addon-links: 5.3.12 => 5.3.12
    @storybook/addons: 5.3.12 => 5.3.12
    @storybook/preset-typescript: 1.2.0 => 1.2.0
    @storybook/react: 5.3.12 => 5.3.12
```

**Additional context**
Should be repeatable with a brand new build of Storybook. I'm using React but I assume it is a general problem? Not using CRA

**Comment by danawoodman** [comment]
Bumppppppppp

**Comment by danawoodman** [comment]
Anyone...?

**Comment by danawoodman** [comment]
Don't forget little ole me!

**Comment by danawoodman** [comment]
Damn you stale bot

**Comment by davidstromback** [comment]
Bump

**Comment by codemonkey800** [comment]
ughh this issue is affecting me too, i'll spend some time looking into it later if no one else is going to 😆

**Comment by aeksco** [comment]
Bump!

This is an almost daily issue for me - even the smallest error during the initial build means I need to completely restart the server, which is growing increasingly time-consuming as our number of stories continues to grow

**Comment by anandau** [comment]
This is really very annoying. Is there any workaround for this?

**Comment by shilman** [comment]
To everyone who is upvoting comments like "bummmp", if you want this fixed, please upvote by adding a 👍 to the issue description. We use the issue upvotes as a signal to help prioritize!

**Comment by ghengeveld** [comment]
Verified. I can reproduce with `@storybook/react@5.2.8`. It appears already resolved in `@storybook/react@6.1.0-alpha.33`, because I cannot reproduce there. For 6.1.0 I changed how the manager and preview get compiled as well as when the server starts (it's basically the first thing that happens now, long before compiling).

@danawoodman could you test with the latest (beta) of 6.1.0?

**Comment by shilman** [comment]
Fixed in 6.1-rc, can you verify @danawoodman? Closing.
